src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -237,16 +239,23 @@ public ResponseEntity<byte[]> getDragAndDropBackgroundFile(@PathVariable Long qu
      * @param dragItemId ID of the drag item, the file belongs to
      * @return The requested file, 403 if the logged-in user is not allowed to access it, or 404 if the file doesn't exist
      */
-    @GetMapping("files/drag-and-drop/drag-items/{dragItemId}/*")
+    @GetMapping("files/drag-and-drop/drag-items/{questionId}/{dragItemId}/*")
     @EnforceAtLeastStudent
-    public ResponseEntity<byte[]> getDragItemFile(@PathVariable Long dragItemId) {
+    public ResponseEntity<byte[]> getDragItemFile(@PathVariable Long questionId, @PathVariable Long dragItemId) {
         log.debug("REST request to get file for drag item : {}", dragItemId);
-        DragItem dragItem = dragItemRepository.findByIdElseThrow(dragItemId);
-        Course course = dragItem.getQuestion().getExercise().getCourseViaExerciseGroupOrCourseMember();
+        DragAndDropQuestion question = quizQuestionRepository.findDnDQuestionByIdOrElseThrow(questionId);
+        DragItem dragItem = question.getDragItems().stream().filter(e -> Objects.equals(e.getId(), dragItemId)).findFirst().orElse(null);
+
+        Course course = question.getExercise().getCourseViaExerciseGroupOrCourseMember();
         authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, null);
+
+        if (dragItem == null) {
+            throw new EntityNotFoundException("Drag item " + dragItemId + " can not found");
```

Comment:
```suggestion
            throw new EntityNotFoundException("Drag item " + dragItemId + " cannot be found");
```

You could also use `orElseThrow(...)` with your exception when you retrieve the `dragItem`.

Reply:
In general, we use `throw new` and I will keep it in this way to preserve consistency.